game/src/client/scenes/raceGame/RaceScene.ts

```diff
@@ -536,6 +525,36 @@ export default class RaceScene extends Phaser.Scene {
 			}
 		}
 	}
+
+	private zoomIn(): void {
+		const cam = this.cameras.main;
+		if (cam.zoom <= this.maxZoom) cam.zoom += 0.1;
+	}
+
+	private zoomOut(): void {
+		const cam = this.cameras.main;
+		if (cam.zoom >= this.minZoom) cam.zoom -= 0.1;
+	}
+
+	private createCameraBounds(x: number, y: number, boardWidth: number, boardHeight: number): void {
+		let pinch = new Pinch(this);
+		pinch.on(
+			"pinch",
+			(pinch) => {
+				var scaleFactor = pinch.scaleFactor;
+				this.cameras.main.zoom *= scaleFactor;
+			},
+			this
+		);
+
+		this.cameras.main.setBounds(
+			x - 200,
+			y - 200,
+			boardWidth * this.distanceBetweenTwoTiles + 600,
+			boardHeight * this.distanceBetweenTwoTiles + 600,
+			true
+		);
```

Comment:
J'ajouterais des constantes pour les 200 et 600. Si je comprends bien leur fonctionnaité, ça pourrait être quelque chose comme :
BOARD_LEFT_CAMERA_MARGIN
BOARD_TOP_CAMERA_MARGIN
BOARD_RIGHT_CAMERA_MARGIN
BOARD_BOTTOM_CAMERA_MARGIN

Éventuellement, on pourrait aussi penser à adapter les margins en fonction du Zoom actuel. Donc ces contantes prendrait peut-être le bord. Adapter les margins en fonction du Zoom nous permettrait d'avoir une caméra centrée quand on est complètement Zoomed Out et que, disons, notre personnage est complètement à gauche du board sans pour autant permettre à la caméra de partir ridiculement loin à gauche quand on est complètement Zoomed In.

On pourrait adapter la caméra bound et les margins de telle sorte que le centre de l'écran ne puisse jamais sortir de la zone du board.